pkg/occlient/volumes.go

```diff
@@ -77,29 +87,57 @@ func (c *Client) AddPVCToDeploymentConfig(dc *appsv1.DeploymentConfig, pvc strin
 	return nil
 }
 
-// UpdatePVCLabels updates the given PVC with the given labels
-func (c *Client) UpdatePVCLabels(pvc *corev1.PersistentVolumeClaim, labels map[string]string) error {
-	pvc.Labels = labels
-	_, err := c.kubeClient.KubeClient.CoreV1().PersistentVolumeClaims(c.Namespace).Update(pvc)
-	if err != nil {
-		return errors.Wrap(err, "unable to remove storage label from PVC")
-	}
-	return nil
-}
-
-// DeletePVC deletes the given PVC by name
-func (c *Client) DeletePVC(name string) error {
-	return c.kubeClient.KubeClient.CoreV1().PersistentVolumeClaims(c.Namespace).Delete(name, nil)
-}
-
 // IsAppSupervisorDVolume checks if the volume is a supervisorD volume
 func (c *Client) IsAppSupervisorDVolume(volumeName, dcName string) bool {
 	return volumeName == getAppRootVolumeName(dcName)
 }
 
+// RemoveVolumeFromDeploymentConfig removes the volume associated with the
+// given PVC from the Deployment Config. Both, the volume entry and the
+// volume mount entry in the containers, are deleted.
+func (c *Client) RemoveVolumeFromDeploymentConfig(pvc string, dcName string) error {
+
+	retryErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+
+		dc, err := c.GetDeploymentConfigFromName(dcName)
+		if err != nil {
+			return errors.Wrapf(err, "unable to get Deployment Config: %v", dcName)
+		}
+
+		volumeNames := getVolumeNamesFromPVC(pvc, dc)
+		numVolumes := len(volumeNames)
+		if numVolumes == 0 {
+			return fmt.Errorf("no volume found for PVC %v in DC %v, expected one", pvc, dc.Name)
+		} else if numVolumes > 1 {
+			return fmt.Errorf("found more than one volume for PVC %v in DC %v, expected one", pvc, dc.Name)
+		}
+		volumeName := volumeNames[0]
+
+		// Remove volume if volume exists in Deployment Config
+		err = removeVolumeFromDC(volumeName, dc)
```

Comment:
This is the meat of the function, IMO. Everything else is leading us up to this step. But it's all in the `RemoveVolumeFromDeploymentConfig` function and makes the function too fat, IMO. I am not insisting on making this function trim by hook or by crook. But it's something we could try to do, no? If not in this PR, we could refactor this in next step when we add interfaces to our code. WDYT?

Reply:
We should remove the function `RemoveVolumeFromDeploymentConfig` as explained here https://github.com/openshift/odo/pull/4226#discussion_r527724180